packages/rolldown/src/utils/transform-to-rollup-output.ts

```diff
@@ -20,6 +21,21 @@ import {
 import { bindingifySourcemap } from '../types/sourcemap'
 import { normalizeErrors } from './error'
 import { transformChunkModules } from './transform-rendered-chunk'
+import { Buffer } from 'node:buffer'
+
+function transformToRollupSourceMap(map: string): SourceMap {
+  const parsed: Omit<SourceMap, 'toString' | 'toUrl'> = JSON.parse(map)
+  const obj: SourceMap = {
+    ...parsed,
+    toString() {
+      return JSON.stringify(obj)
```

Comment:
Here could be directly using `map`

Reply:
Doing that will break this test.
https://github.com/rolldown/rolldown/pull/3064/files#diff-effa1bb20ad69d9982696be2e4798f5ef323ae4129a0461a45ae63e69c331e64R20-R27
I'm not sure if people would do this though.
We are not going to support that case, I think we should mark all properties in `SourceMap` as `readonly`.

Reply:
Get it.